rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/DisconnectionRouter.java

```diff
@@ -17,26 +17,19 @@
  * A class that is responsible for routing all potential sources of disconnection to an Observable that emits only errors.
  */
 @ConnectionScope
-class DisconnectionRouter {
+class DisconnectionRouter implements DisconnectionRouterInput, DisconnectionRouterOutput {
 
-    private final PublishRelay<BleException> disconnectionErrorRelay = PublishRelay.create();
-
-    private final Observable disconnectionErrorObservable;
+    private final PublishRelay<BleException> disconnectionErrorInputRelay = PublishRelay.create();
+    private final Observable<BleException> disconnectionErrorOutputObservable;
 
     @Inject
     DisconnectionRouter(
             @Named(DeviceModule.MAC_ADDRESS) final String macAddress,
             final RxBleAdapterWrapper adapterWrapper,
             final Observable<RxBleAdapterStateObservable.BleAdapterState> adapterStateObservable
     ) {
-        disconnectionErrorObservable = Observable.merge(
-                disconnectionErrorRelay
-                        .flatMap(new Func1<BleException, Observable<?>>() {
-                            @Override
-                            public Observable<?> call(BleException e) {
-                                return Observable.error(e);
-                            }
-                        }),
+        disconnectionErrorOutputObservable = Observable.merge(
+                disconnectionErrorInputRelay,
                 adapterStateObservable
```

Comment:
Shouldn't it rely on the initial state as well?

Reply:
Was not an issue till now. The DisconnectionRouter would not get created (@ConnectionScope) if the adapter is off

Reply:
ok